File: javascript01/step11/test02_2.js
/* DBMS에 SQL명령 보내기 2) insert : 데이터 추가 명령
*/

const mysql = require('mysql')

var connection = mysql.createConnection({
  user     : 'webapp',
  password : '1111',
  database : 'webappdb'
})

connection.connect()

// 순서 상관없다.
// ' + ' 는 줄바꿈하기 위함
// 비번은 암호화
connection.query('insert into memb(name, email, tel, pwd)' +
  ' values("aaa2", "aaa2@test.com", "111-1111", password("1111"))', function (error, results) { // error가 안들어오면 undefined
  console.log('결과를 가져왔음')
  if (error) throw error

  console.log(results)
})

connection.end()
console.log('MySQL 서버와의 연결 끊기가 예약되었음')

File: java01/src/step20/Test02_2.java
/* JDBC 프로그래밍: Statement 객체 - INSERT 실행
 * => Statement.executeUpdate()
 *    - INSERT, UPDATE, DELETE 문 등 DML, DDL을 실행한다. 
 */
package step20;

import java.sql.DriverManager;

public class Test02_2 {

  public static void main(String[] args) throws Exception {

    com.mysql.jdbc.Driver mysqlDriver = new com.mysql.jdbc.Driver();

    DriverManager.registerDriver(mysqlDriver);

    java.sql.Connection con = DriverManager.getConnection(
        "jdbc:mysql://localhost:3306/webappdb", 
        "webapp", 
        "1111"); 

    java.sql.Statement stmt = con.createStatement();

    int count = stmt.executeUpdate("insert into memb(name,tel,email,pwd) " 
        + " values('홍길동', '1212-1111', 'hong100@text.com', password('1111'))");
    
    System.out.println(count);

    stmt.close();

    con.close();

  }

}





